Async/Await

tjgeni opened the pull request:
Async/Await done please review it @yoga1234 . thanks

Commit:
translating into indonesian

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/javascript-tutorial/id.javascript.info?pullRequest=129) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/javascript-tutorial/id.javascript.info?pullRequest=129) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: tjgeni<br/>:x: tegarjgap<br/><hr/>**tegarjgap** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/javascript-tutorial/id.javascript.info?pullRequest=129) it.</sub>

tjgeni committed:
Merge branch 'master' into AsyncAwait

yoga1234 commented on 1-js/11-async/08-async-await/02-rewrite-async-2/solution.md:
```diff
@@ -19,28 +18,26 @@ async function loadJson(url) {
+    } catch (err) {
       if (err instanceof HttpError && err.response.status == 404) {
-        // loop continues after the alert
+        // loop melanjutkan setelah alert
```
apa yang ini mungkin dilanjutkan/berlanjut ? :/

yoga1234 commented on 1-js/11-async/08-async-await/article.md:
```diff
@@ -1,20 +1,20 @@
+Ada sintaksis spesial untuk bekerja dengan promises dengan cara yang lebih nyaman, dipanggil "async/await". Ini sangat mudah dipahami dan digunakan.
 
-## Async functions
+## Function Async
```
Fungsi Async?

yoga1234 commented on 1-js/11-async/08-async-await/article.md:
````diff
@@ -34,20 +34,21 @@ async function f() {
 
 ```js
-// works only inside async functions
+// bekerja hanya di dalam function async
````
Fungsi?

yoga1234 commented:
Disini masih banyak kata yang pake function gan, mau diubah atau gimana ya? hehe

yoga1234 commented:
Sekalian cek issue baru dong.. hahaha

tegarjgap commented:
> Disini masih banyak kata yang pake function gan, mau diubah atau gimana ya? hehe

kalo menurut saya, karena "function" ini istilah khusus gitu mas Yoga., jadi ga di terjemahin ke indonesia kalo emang mau diubah gapapa haha

tegarjgap commented:
> > Disini masih banyak kata yang pake function gan, mau diubah atau gimana ya? hehe
> 
> kalo menurut saya, karena "function" ini istilah khusus gitu mas Yoga., jadi ga di terjemahin ke indonesia kalo emang mau diubah gapapa haha

oh lebih baik kalo function jadi fungsi deh.

yoga1234 commented on 1-js/11-async/08-async-await/article.md:
```diff
@@ -224,81 +226,82 @@ async function f() {
   } catch(err) {
 *!*
-    alert(err); // TypeError: failed to fetch
+    alert(err); // TypeError: gagal untuk mengambil
```
Errornya jangan diterjemahin gan, kasih catatan tambahan aja TypeError: failed to fetch (gagal mengambil resource). Soalnya itu keluaran dari Javascriptnya... hehe

tegarjgap replied:
siap mas. astaga lupa. saya update dulu

yoga1234 replied:
Mantap, santai

yoga1234 commented on 1-js/11-async/08-async-await/article.md:
```diff
@@ -224,81 +226,82 @@ async function f() {
+// f() menjadi sebuah promise yang direject
 *!*
-f().catch(alert); // TypeError: failed to fetch // (*)
+f().catch(alert); // TypeError: gagal untuk mengambil // (*)
```
Yang ini.. hehe

tegarjgap replied:
okayy

yoga1234 commented:
Soalnya penerjemah lain pakenya fungsi sih, tapi santai aja sih. Yang lain kayanya belum nimbrung di issue baru. hehe
Sama tadi belum saya komen ada penggunaan _top-level code_ tapi dibawahnya jadi "kode top level"

yoga1234 reviewed (changes requested):
Ada beberapa komentar gan..

tjgeni committed:
Update penerjemahan

ganti kata "melanjutkan" menjadi "dilanjutkan"

tegarjgap commented:
> top-level code

ini diterjemahin jadi kode tingkat atas / tetap inggris ya mas sebaiknya?

yoga1234 commented:
> > top-level code
> 
> ini diterjemahin jadi kode tingkat atas / tetap inggris ya mas sebaiknya?

Boleh yang mana aja dulu, tar paling ubah pas rules penulisan udah ada.. hehehe

tjgeni committed:
update penerjemahan

ubah kata "function" menjadi "fungsi" + "kode top-level" menjadi "top-level code"

tegarjgap commented:
siap. done revisi nya mas @yoga1234 . mintol dicek lagi, hehe thankyou..

yoga1234 commented:
> siap. done revisi nya mas @yoga1234 . mintol dicek lagi, hehe thankyou..

Mantap, buat bot nya kasih komen /done gan..

yoga1234 commented on 1-js/11-async/08-async-await/article.md:
````diff
@@ -259,15 +259,15 @@ async function f() {
 ```smart header="`async/await`dan`promise.then/catch`" Ketika kita menggunakan `async/await`, kita jarang membutuhkan `.then`, karena `await`menangani waiting tersebut untuk kita. Dan kita dapat menggunakan sebuah `try..catch` biasa dibandingkan`.catch`. Itu biasanya (tidak selalu) lebih nyaman.
 
-Tetapi pada kode top level, saat kita berada di luar function `async`, kita secara sintaks tidak dapat menggunakan `await`, jadi itu sebuah latihan normal untuk menambah `.then/catch` untuk menangani hasil akhir atau error yang terjatuh.
+Tetapi pada *top-level code*, saat kita berada di luar fungsi `async`, kita secara sintaks tidak dapat menggunakan `await`, jadi itu sebuah latihan normal untuk menambah `.then/catch` untuk menangani hasil akhir atau error yang terjatuh.
````
Yang ini lebih enak, "untuk menangani hasil akhir atau jika terjadi error" ?

tegarjgap replied:
yaps. lebih enak sperti itu mas @yoga1234 . thanks. sudah saya update kembali

tjgeni committed:
update article.md

perbaikan kata "terjatuh"

tegarjgap commented:
> Please make the requested changes. After it, add a comment "/done".
> Then I'll ask for a new review 👻

/done

yoga1234 reviewed (approved):
Mantap gan, semangat!!!
